### ISSUE TITLE
Issues when installing new version from 24-11-2023 

### ISSUE THREAD

**Issue by JWBro** [LMS-Community/slimserver#945]
I run the slimserver on a number of PC's. With the new version every installation seems to have issues when overwriting the following files in the directories: server\Graphics, server\HTML\Classic\html\images, server\HTML\Default\html\images, server\HTML\EN\html\images,. Tried to remove the read-only status from these files BUT that does not work.
I have uninstalled the previous version of the slimserver and had to start to completely rescan the music libraries. However I have large music libraries (one with over 11000 albums) and that takes a lot of time.
This issues also happened in the last year a couple of times. And by the way I observed that there is every other day almost a new version of teh program; this in combination of what described it's pretty annoying.

**Comment by mherger** [comment]
Unfortunately you don't give any information about what you are talking about... I must assume you are running LMS 8.4 on Windows? 8.4 is still under development. Hiccups must be expected. If you don't like the regular updates, stay with the stable 8.3.1.

As for the actual issue: not sure what's happening there. Are you saying that some files are read-only and can't be overwritten by the installer? Would you get an error message?

**Comment by JWBro** [comment]
Yes I do use (and run) LMS 8.4.0 on both Windows 10 and 11. The version is 8.4.0-1700807977 dated 20231124. The issues of not being able to overwrite the various image and of font files: 
server\Graphics: all bmp, font-files
server\HTML\Classic\html\images: GIF files
server\HTML\Default\html\images: GIF files
server\HTML\EN\html\images: PNG and GIF files
Using File Explorer and properties for the various directories these files are read-only; Tried to remove the read-only flag for the files in these folders but no luck; the LMS installer still reports that the files can not be overwritten. See the screenshot here:
![2023-11-24_14-48](https://github.com/Logitech/slimserver/assets/66782406/53350bb2-d077-498c-a11e-a7de970e0c81)

What's also happening that I have to maually 'kill' the various running LMS executables as the new installer can not do this by itself. No big issue, one always can kill it using the task manager. 
Sure one might expect hickups with SW that is under development, that is a given I understand that very well, as I have been a SW developper myself.
Hop ethis helps
regards 
Jan

**Comment by JWBro** [comment]
I had another try with yet another PC with the previous version of LMS 8.4; once more the issues came when the installer tried to overwrite the graphics files (BMP, GIF, PNG etc). I had to to skip overwriting the graphic files, but there are so many. So I cancelled the update and instead I removed from the Squeezebox\server folder the sub folders Graphics, HTML and Slim. Than I restarted the instalation and it came to and and .... I checked the files in the earlier removed folders versu another fresh installation on a PC that had the new version of LMS. It seems OK in bith number of folders & files and size. Than I ran this new installtion to see whether it was capable to scan the libraries and to play the music. This worked OK.
Looks to me that with the new installer bumps into illegal file overwrites (although all the files had the read only removed. Hope this helps identifying the issue at hand.

**Comment by michaelherger** [comment]
Thanks for the report! I've had at least one more user report a similar issue. I'm currently going through the InnoSetup changelog to see whether something might have changed which has an impact on this. But the installer should run in admin mode, in which case it should work just fine...

Did you get the screen from Windows asking for confirmation that you want to run the installer?

**Comment by JWBro** [comment]
I used the installer always in admin mode; also in his mode it refuses to verwrite the graphics files.

Yes have seen the screen from Windows to confirm to run the installer.

best regards
Jan

**Comment by JWBro** [comment]
Hello,

Update:

-          I tried to update LMS 8.4.0 using the popup of LMS little windows that in the righthand bottom corner ‘install new version’ That does not work as it is not running with admin rights.

-          The only way to update LMS 8.4.0 is to download the new version from the github site, stopping the running LLMS server and than to run the LMS installer with admin rights.
Best regards
Jan Broenink

**Comment by michaelherger** [comment]
May I ask you which version of the Windows installer you're using? There's currently an _experimental_ 64-bit version in addition to the previous 32-bit version. Please note that the former is, as its name says, still experimental. It's work in progress. So if you want to have a stable system, make sure you're using the 32-bit version.

Both version should _always_ require admin mode.

**Comment by JWBro** [comment]
Hi, the installer that I use the the 32b version (it install the LMS into the folder ProgramFiles (x86).
I downloaded the LMS installer from downloads/slimdevices.com/nightly/?ver=8.4
I have seen that you’re working on a 64 bits version with Strawberry perl version 5.32.1.1 64b
I am familiar with the Strawberry Perl version that I have installed but that is version 5.38.0.1 64b; I assume that you 64b LMS will also work with the later version Perl. Is my assumption correct?

Note: I use several PC’s and servers the 32 b LMS version 8.4.0 (latest) runs fine on Windows 10, Windows 11 and Windows Server 2019 as well as 2022 (all tested).
If the 64b version LMS wil work with the 5.38.0.1 64 b, I will test it on a machine with a small music library. No problem and will report my findings to you.

Best regards from the Netherlands
Jan

**Comment by michaelherger** [comment]
> I am familiar with the Strawberry Perl version that I have installed but that is version 5.38.0.1 64b; I assume that you 64b LMS will also work with the later version Perl. Is my assumption correct?

Unfortunately no. We have so many binaries which we ship with LMS because they're not available yet from other repositories, you'd have to build them all yourself. And I doubt that's something you'd want to do. Yes, there's overhead using LMS own installation of Perl. But I can't handle it differently as long as I want to keep job and family...

**Comment by JWBro** [comment]
Hi Michael,

I surely understand your point, no doubt about that ( I have the experience to develop software myself).

Anyway I have installed the LMS 64b version as of 20231128 (aka version 8.4.0 – 170116362).
The PC had already the Strawberry Perl version 5.38.0.1 64b; one of the first action after I started the LMS installer that it found this very Perl version and copied it to the folder Program Files\Squeezebox. When the install was completed I tried to start SqzSvcMgr.exe; it did indeed accomplisehd some task but coul not start LMS. I checked the Starwberry folder of my PC versus the Perl folder in Program Files\Squeezebox; it turned out that about 60% of the original installed Stawberry Perl was copied; so I copied the content of my Starwberry Perl installtion to the Perl folder in Squeezebox. Than I started the SqzSvcMgr.exe and LMS started and runs fine as I could select and start the music …… Than I opted for a larger library with known content, it was scanned and ran OK (the library size was 558 albums).

OK if I can be of anymore help in testing software, let me know. Thanks for the good work !!!!

Best regards
Jan
